js/main.js

```diff
@@ -162,11 +162,81 @@ class MainPresenter {
    */
   _onAutoRefresh() {
     const self = this;
-    if (!self._hasCompilationErrors) {
+    if (!self._hasCompilationErrors && self._shouldAutoRun()) {
       self._onBuild(true, false, true);
     }
   }
 
+  /**
+   * Checks if auto-run should be executed based on tab and checkbox state.
+   *
+   * @returns {boolean} True if auto-run should execute, false otherwise.
+   * @private
+   */
+  _shouldAutoRun() {
+    const self = this;
+    return self._isOnCodeEditorTab() && self._isAutoRunEnabled();
+  }
+
+  /**
+   * Checks if the user is currently on the code editor tab.
+   *
+   * @returns {boolean} True if on code editor tab, false otherwise.
+   * @private
+   */
+  _isOnCodeEditorTab() {
+    const codeEditorPane = document.getElementById("code-editor-pane");
+    return codeEditorPane && codeEditorPane.getAttribute("aria-hidden") !== "true";
+  }
+
+  /**
+   * Checks if the auto-run checkbox is checked.
+   *
+   * @returns {boolean} True if auto-run is enabled, false otherwise.
+   * @private
+   */
+  _isAutoRunEnabled() {
+    const autoRunCheck = document.getElementById("auto-run-check");
+    return autoRunCheck && autoRunCheck.checked;
+  }
+
+  /**
+   * Shows a message indicating no results were produced.
+   *
+   * @private
+   */
+  _showNoResultsMessage() {
+    const self = this;
+    const resultsSection = document.getElementById("results");
+    resultsSection.style.display = "block";
+
+    // Clear existing content
+    const existingMessage = resultsSection.querySelector(".no-results-message");
+    if (existingMessage) {
+      existingMessage.remove();
+    }
+
+    // Create and insert no results message
```

Comment:
Could this div instead be in `index.html` but with display of none by default? That would mean this code simply has to show or hide that div instead of recreating / removing it each time. Also, the style defined here could go into `/style/style.css` instead.